stravalib/client.py

```diff
@@ -715,38 +764,28 @@ def upload_activity(self, activity_file, data_type, name=None, description=None,
                 raise TypeError("Invalid type specified for activity_file: {0}".format(type(activity_file)))
 
         valid_data_types = ('fit', 'fit.gz', 'tcx', 'tcx.gz', 'gpx', 'gpx.gz')
-        if not data_type in valid_data_types:
-            raise ValueError("Invalid data type {0}. Possible values {1!r}".format(data_type, valid_data_types))
+        if data_type not in valid_data_types:
+            raise ValueError(f'Invalid data type {data_type}. Possible values {valid_data_types!r}')
 
         params = {'data_type': data_type}
         if name is not None:
             params['name'] = name
         if description is not None:
             params['description'] = description
         if activity_type is not None:
-            warnings.warn(
```

Comment:
just curious is the activity_type no longer undocumented?

Reply:
No, I moved the warning after the check for valid activity types. That ValueError should be raised anyway in case of an invalid type. This move allowed me to reduce complexity of writing the tests (because I only need to check if either an exception or warning is raised, but not both).